geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatSerialization.scala

```diff
@@ -0,0 +1,265 @@
+/***********************************************************************
+  * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0 which
+  * accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import java.nio.ByteBuffer
+import java.util.Date
+
+import com.google.common.primitives.Bytes
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Stats are serialized as a byte array where the first byte indicates which type of stat is present.
+ * The next four bits contain the size of the serialized information.
+ * A SeqStat is serialized the same way, with each individual stat immediately following the previous in the byte array.
+ */
+object StatSerialization {
+  // bytes indicating the type of stat
+  val MINMAX_BYTE: Byte     = '0'
+  val ISC_BYTE: Byte        = '1'
+  val EH_BYTE: Byte         = '2'
+  val RH_BYTE: Byte         = '3'
+
+  /**
+   * Fully serializes a stat by formatting the byte array with the "kind" byte
+   *
+   * @param kind byte indicating the type of stat
+   * @param bytes serialized stat
+   * @return fully serialized stat
+   */
+  private def serializeStat(kind: Byte, bytes: Array[Byte]): Array[Byte] = {
+    val size = ByteBuffer.allocate(4).putInt(bytes.length).array
+    Bytes.concat(Array(kind), size, bytes)
+  }
+
+  protected [stats] def packMinMax(mm: MinMax[_]): Array[Byte] = {
+    serializeStat(MINMAX_BYTE, s"${mm.attrIndex};${mm.attrType};${mm.min};${mm.max}".getBytes)
+  }
+
+  protected [stats] def unpackMinMax(bytes: Array[Byte]): MinMax[_] = {
+    val split = new String(bytes).split(";")
+    require(split.size == 4)
+
+    val attrIndex = split(0).toInt
+    val attrTypeString = split(1)
+    val min = split(2)
+    val max = split(3)
+
+    val attrType = Class.forName(attrTypeString)
+    attrType match {
+      case _ if attrType == classOf[Date] =>
```

Comment:
[style] same as noted above, in general you don't want to use a match if you're not actually extracting anything. (sometimes it can look cleaner, but not really in this case...)